core/utils/firewall/firewall.go

```diff
@@ -0,0 +1,50 @@
+package firewall
+
+import (
+	"fmt"
+
+	"github.com/1Panel-dev/1Panel/core/utils/cmd"
+)
+
+func UpdatePort(oldPort, newPort string) error {
+	firewalld := cmd.Which("firewalld")
+	if firewalld {
+		status, _ := cmd.Exec("LANGUAGE=en_US:en firewall-cmd --state")
+		isRunning := status == "running\n"
+		if isRunning {
+			return firewallUpdatePort(oldPort, newPort)
+		}
+	}
+
+	ufw := cmd.Which("ufw")
+	if !ufw {
+		return nil
+	}
+	status, _ := cmd.Exec("LANGUAGE=en_US:en ufw status | grep Status")
+	isRuning := status == "Status: active\n"
+	if isRuning {
+		return ufwUpdatePort(oldPort, newPort)
+	}
+	return nil
+}
+
+func firewallUpdatePort(oldPort, newPort string) error {
+	stdout, err := cmd.Execf("firewall-cmd --zone=public --add-port=%s/tcp --permanent", newPort)
+	if err != nil {
+		return fmt.Errorf("add (port: %s/tcp) failed, err: %s", newPort, stdout)
+	}
+
+	_, _ = cmd.Execf("firewall-cmd --zone=public --remove-port=%s/tcp --permanent", oldPort)
+	_, _ = cmd.Exec("firewall-cmd --reload")
+	return nil
+}
+
+func ufwUpdatePort(oldPort, newPort string) error {
+	stdout, err := cmd.Execf("ufw allow %s", newPort)
+	if err != nil {
+		return fmt.Errorf("add (port: %s/tcp) failed, err: %s", newPort, stdout)
+	}
+
+	_, _ = cmd.Execf("ufw delete allow %s", oldPort)
+	return nil
+}
```

Comment:
There are no major issues with the given snippet. The code seems to correctly uses `cmd` for executing shell commands on different systems based on which package name they have chosen.

However, minor improvements like considering the return types of the executables (`return int`, not returning an error), and using a const variable rather than hardcoding values for the ports might be beneficial:

```go
// Define constants here if needed

const (
	oldPort  = "80" // Change to your old port here
	newPort  = "443" // Example new port from above example
	... // rest similar
)

func UpdatePort(oldPort string, newPort string) error {
	switch f := &foo{nil}: {f.func1(); }
	return fooUpdateNewOrOldPort(oldPort, newPort)
}

```

This small change could make the function more readable since it doesn't rely on hard-coded values and can easily maintain multiple ports. Also, this kind of refactoring helps prevent future errors related to changing variables.